crates/conductor_lib/src/context_builder.rs

```diff
@@ -175,6 +184,9 @@ impl ContextBuilder {
         let eav_storage = self
             .eav_storage
             .unwrap_or_else(|| Arc::new(RwLock::new(EavMemoryStorage::new())));
+        let metric_publisher = self
+            .metric_publisher
+            .unwrap_or_else(|| Arc::new(RwLock::new(DefaultMetricPublisher::new())));
```

Comment:
if you derive a default implementation for DefaultMetricPublisher you might be able to get away with unwrap_or_default()